src/output/dnssim/common.c

```diff
@@ -60,6 +110,13 @@ void _output_dnssim_create_request(output_dnssim_t* self, _output_dnssim_client_
         goto failure;
     }
 
+    req->question = req->dns_q->at;
+    req->question_len = parse_qsection(req->dns_q);
+    if (req->question_len < 0) {
```

Comment:
```
../../src/output/dnssim/common.c:115: warning: comparison is always false due to limited range of data type
```

Reply:
You're putting a `ssize_t` into a `uint16_t`, this will turn -1 into 0xffff.

You have to either change `question_len` to a signed value or change how you return "not found"/"error" from `parse_qsection()`.